ostd/src/mm/page_table/mod.rs

```diff
@@ -380,4 +386,43 @@ pub trait PageTableEntryTrait:
     /// The level of the page table the entry resides is given since architectures
     /// like amd64 only uses a huge bit in intermediate levels.
     fn is_last(&self, level: PagingLevel) -> bool;
+
+    /// Converts the PTE into its corresponding `usize` value.
+    fn as_usize(self) -> usize {
+        // SAFETY: `Self` is Copy and has the same memory representation as usize
+        unsafe { transmute_unchecked(self) }
+    }
+
+    /// Converts a usize `pte_raw` into a PTE.
+    ///
+    /// # Safety
+    ///
+    /// The `pte_raw` must represents a valid PTE.
+    unsafe fn from_usize(pte_raw: usize) -> Self {
+        // SAFETY: `Self` is Copy and has the same memory representation as usize
```

Comment:
```suggestion
        // SAFETY: `Self` is `Pod` and has the same memory representation as `usize`.
```

`Copy` is not enough.

For example, transmutation of pointers to `usize` is explicitly [warned against](https://doc.rust-lang.org/std/mem/fn.transmute.html#transmutation-between-pointers-and-integers) in the Rust documentation.[^1]

[^1]: I hope (and assume) we don't implement `Pod` for `*mut T`/`*const T` now.